kratos/tests/test_variable_redistribution.py

```diff
@@ -116,91 +97,161 @@ def ReferenceSolution(node,variable):
+            self.interface_model_part,
+            PRESSURE,
+            TEMPERATURE)
 
-            VariableRedistributionUtility.ConvertDistributedValuesToPoint(
-                self.interface_model_part,
-                PRESSURE,
-                TEMPERATURE)
+        for cond in self.model_part.Conditions:
+            area = cond.GetGeometry().Area()
+            for node in cond.GetNodes():
+                nodal_area = node.GetSolutionStepValue(NODAL_PAUX)
+                node.SetSolutionStepValue(NODAL_PAUX,nodal_area+area/3.0)
 
-            for cond in self.model_part.Conditions:
-                area = cond.GetGeometry().Area()
-                for node in cond.GetNodes():
-                    nodal_area = node.GetSolutionStepValue(NODAL_PAUX)
-                    node.SetSolutionStepValue(NODAL_PAUX,nodal_area+area/3.0)
+        if self.print_output:
+            self.InitializeOutput()
+            self.PrintOutput()
+            self.FinalizeOutput()
 
-            if self.print_output:
-                self.InitializeOutput()
-                self.PrintOutput()
-                self.FinalizeOutput()
+        self.CheckDoubleResults(self.interface_model_part, TEMPERATURE, NODAL_PAUX)
 
-            self.CheckDoubleResults(TEMPERATURE,NODAL_PAUX)
 
     def SetUpProblem(self):
-        current_model = Model()
-        self.model_part = current_model.CreateModelPart("Model")
-        self.model_part.AddNodalSolutionStepVariable(FLAG_VARIABLE)
-        self.model_part.AddNodalSolutionStepVariable(PRESSURE)
-        self.model_part.AddNodalSolutionStepVariable(NODAL_PAUX)
-        self.model_part.AddNodalSolutionStepVariable(TEMPERATURE)
-        self.model_part.AddNodalSolutionStepVariable(MAPPER_SCALAR_PROJECTION_RHS)
-        self.model_part.AddNodalSolutionStepVariable(VELOCITY)
-        self.model_part.AddNodalSolutionStepVariable(VORTICITY)
-        self.model_part.AddNodalSolutionStepVariable(ACCELERATION)
-        self.model_part.AddNodalSolutionStepVariable(MAPPER_VECTOR_PROJECTION_RHS)
-
-        model_part_io = ModelPartIO(self.input_file)
-        model_part_io.ReadModelPart(self.model_part)
-
-        self.model_part.SetBufferSize(1)
+        with UnitTest.WorkFolderScope(self.work_folder,__file__):
+            current_model = Model()
+            self.model_part = current_model.CreateModelPart("Model")
+            self.model_part.AddNodalSolutionStepVariable(FLAG_VARIABLE)
+            self.model_part.AddNodalSolutionStepVariable(PRESSURE)
+            self.model_part.AddNodalSolutionStepVariable(NODAL_PAUX)
+            self.model_part.AddNodalSolutionStepVariable(TEMPERATURE)
+            self.model_part.AddNodalSolutionStepVariable(VELOCITY)
+            self.model_part.AddNodalSolutionStepVariable(VORTICITY)
+            self.model_part.AddNodalSolutionStepVariable(ACCELERATION)
+
+            model_part_io = ModelPartIO(self.input_file)
+            model_part_io.ReadModelPart(self.model_part)
```

Comment:
May be we can use 'testing.utilities.ReadModelPart` here? This would be beneficial if want to use the same tests in mpi as well :)

Reply:
OK! I'll try.

Reply:
Done.